### PR TITLE
server: replace blog prefix separator from ':' to '-'

### PR DESCRIPTION
**Pull request by bmizerany**
This fixes issues with blob file names that contain ':' characters to be
9 rejected by file systems that do not support them.

### CODE REVIEW

**Review by mxyng — APPROVED** [review]
🚢

File: server/fixblobs_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	"strings"
+	"testing"
+)
+
+func TestFixBlobs(t *testing.T) {
+	cases := []struct {
+		path string
+		want string
+	}{
+		{path: "sha256:1234", want: "sha256-1234"},
```
Comment (pdevine) [inline server/fixblobs_test.go:18]: maybe a negative test case too?

File: server/images.go [hunk 2]
```diff
@@ -795,10 +795,8 @@ func PruneLayers() error {
 
 	for _, blob := range blobs {
 		name := blob.Name()
-		if runtime.GOOS == "windows" {
-			name = strings.ReplaceAll(name, "-", ":")
-		}
-		if strings.HasPrefix(name, "sha256:") {
+		name = strings.ReplaceAll(name, "-", ":")
+		if strings.HasPrefix(name, "sha256-") {
```
Comment (pdevine) [inline server/images.go:799]: I think this would still be `if strings.HasPrefix(name, "sha256:") {`

**Review by pdevine — COMMENTED** [review]
Looks good. The only real comment is the one in `server/images.go` which I think would result in skipping all of the blobs.

For the tests, did you want to add in 2+ files in dir? I added the comment about having a negative test case too.

**Review by pdevine — APPROVED** [review]

**Pull request merged**